include/mbedtls/config_psa.h

```diff
@@ -147,12 +147,15 @@ extern "C" {
 #endif
 
 #if defined(PSA_WANT_ALG_JPAKE)
+#if !defined(MBEDTLS_PSA_ACCEL_ALG_JPAKE)
 #define MBEDTLS_PSA_BUILTIN_PAKE 1
 #define MBEDTLS_PSA_BUILTIN_ALG_JPAKE 1
 #define MBEDTLS_ECP_DP_SECP256R1_ENABLED
 #define MBEDTLS_BIGNUM_C
 #define MBEDTLS_ECP_C
 #define MBEDTLS_ECJPAKE_C
+#define MBEDTLS_SHA256_C
```

Comment:
I don't understand why `MBEDTLS_SHA256_C` was added here in the first place. EC J-PAKE is not tied to SHA-256 and can use any hash. That's true not only of the abstract algorithm, but also of the PSA API and of the legacy API. The only part of `ecjpake.c` that's tied to SHA-256 is the self-test function, which compiles to an empty function if SHA-256 is not defined.

This is hardly related to the on-going driver-only hashes work. As of Mbed TLS 3.3, `ecjpake.c` works when all hashes, including SHA-256, are accelerated. So, it must keep working. It would be the same if we weren't re-restructuring how driver-only hashes are handled.

(What depends on SHA-256 is `MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED` but that's a different thing. Also, most if not all test cases depend on SHA-256. But I don't think that's a reason to add `MBEDTLS_SHA256_C` here.)

Reply:
In psa_crypto_pake.c:psa_pake_setup we claim that our EC J-PAKE implementation only support SECP_R1 and SHA-256 (I did not check this is true). Thus here I was just me being confused about the current limitations of our EC J-PAKE implementation and the dependencies for EC J-PAKE in general. Anyway, we are removing it.